Modules/SubModule_Import/xASL_imp_DCM2NII_Subject.m

```diff
@@ -146,17 +149,33 @@
     if x.modules.import.settings.bUseVisits
         if ~isempty(imPar.visitNames{iVisit}) && strcmp(imPar.visitNames{iVisit}(1),'_')
             % Only add '_' if there isn't one already
-            SubjDir = fullfile(imPar.TempRoot, [subjectID imPar.visitNames{iVisit}]);
+            SubjDir = fullfile(imPar.TempRoot, [subjectExport imPar.visitNames{iVisit}]);
         else
             % Subject/session directory with '_'
-            SubjDir = fullfile(imPar.TempRoot, [subjectID '_' imPar.visitNames{iVisit}]);
+            SubjDir = fullfile(imPar.TempRoot, [subjectExport '_' imPar.visitNames{iVisit}]);
         end
     else
-        SubjDir = fullfile(imPar.TempRoot, subjectID);
+        SubjDir = fullfile(imPar.TempRoot, subjectExport);
     end
 
 end
 
 
+%% If a subject ID has a `_` or a `-` or something else, we need to get rid of it
+function subjectExport = xASL_imp_SubjectName(subjectID)
+
+    % If a subject ID has a `_` or a `-` or something else, we need to get
+    % rid of it in the directory name for BIDS, for the import of the
+    % sourcedata we should still know about that though, which is why we
+    % use a new variable for the subject ID directory export.
+    
+    subjectExport = xASL_adm_CorrectName(subjectID, 2);
+    
+    if ~strcmp(subjectID,subjectExport)
+        fprintf(2,'Special characters in subject ID...\n');
```

Comment:
It is really helpful for the user to print something like:
Special characters in subject ID, changing 'subject_name' -> 'subjectname'

Reply:
Good idea, will do 👍